packages/app/src/debug/DebugPageHeader.vue

```diff
@@ -112,40 +112,49 @@ import CommitIcon from '~icons/cy/commit_x14'
 import { IconTimeStopwatch } from '@cypress-design/vue-icon'
 import { gql } from '@urql/core'
 import { dayjs } from '../runs/utils/day.js'
+import { useI18n } from 'vue-i18n'
+
+const { t } = useI18n()
 
-// runNumber and commitHash dont currently exist in the query and therefore are being obtained as props instead
 gql`
 fragment DebugPage on CloudRun {
-	id
-	createdAt
-	status
+  id
+  runNumber
+  createdAt
+  status
   totalDuration
+  commitInfo {
+    sha
+  }
   url
-  tags {
-    id
-    name
+  ...RunResults
+  commitInfo {
+    authorName
+    summary
+    branch
   }
-	...RunResults
-	commitInfo {
-		authorName
-		authorEmail
-		summary
-		branch
-	}
 }
 `
 
 const props = defineProps<{
   gql: DebugPageFragment
-  commitsAhead: string
-  commitHash: string
-  runNumber: number
+  commitsAhead: number
 }>()
 
 const debug = computed(() => props.gql)
 
 const relativeCreatedAt = computed(() => dayjs(new Date(debug.value.createdAt!)).fromNow())
 
+/*
+  Format duration to in HH:mm:ss format. The `totalDuration` field is milliseconds. Remove the leading "00:" if the value is less
+  than an hour. Currently, there is no expectation that a run duration will be greater 24 hours or greater, so it is okay that
+  this format would "roll-over" in that scenario.
+  Ex: 1 second which is 1000ms = 00:01
+  Ex: 1 hour and 1 second which is 3601000ms = 01:00:01
+*/
+
+const totalDuration = computed(() => dayjs.duration(debug.value.totalDuration!).format('HH:mm:ss').replace(/^0+:/, ''))
```

Comment:
We have the same function [here](https://github.com/cypress-io/cypress/blob/eb93a7963bad41c8c59595eebe363fb2e16edc94/packages/app/src/runs/RunCard.vue#L123). Would be nice to create a utility method.

Reply:
Made a composable to encapsulate the logic